### PR TITLE
[ECO-2056] Add e2e search test

### PR DESCRIPTION
**Pull request by CRBl69**
# Description

This PR adds an e2e test for the search feature on the home page.

It also fixes may things with the e2e configuration, that will allow the writing of other e2e tests much faster.

# Testing

Ensure CI passes.

### CODE REVIEW

File: src/typescript/sdk/tests/post-test.ts [hunk 1]
```diff
@@ -2,5 +2,5 @@
 import { DockerTestHarness } from "../src/utils/test/docker/docker-test-harness";
 
 export default async function postTest() {
-  await DockerTestHarness.stop();
+  await DockerTestHarness.stop(false);
```
Comment (xbtmatt) [inline src/typescript/sdk/tests/post-test.ts:5]: I think we should make this a named variable or make the function name clearer here. I can't remember what the boolean value does when calling `stop`

It should probably just be two different functions with the functionality conveyed in the function name, and then remove the boolean arg. Easier to read that way

**Review by xbtmatt — COMMENTED** [review]
Looks good so far, just a couple small changes

File: src/typescript/package.json [hunk 2]
```diff
@@ -25,14 +25,13 @@
     "dev:debug": "pnpm dotenv -v FETCH_DEBUG=true -- pnpm run dev",
     "dev:debug-verbose": "pnpm dotenv -v FETCH_DEBUG_VERBOSE=true -- pnpm run dev",
     "down": "pnpm run prune",
-    "e2e:frontend": "pnpm run load-env:e2e-frontend -- turbo run e2e:frontend --filter @econia-labs/emojicoin-frontend --log-prefix none",
+    "e2e:frontend": "turbo run e2e:frontend --filter @econia-labs/emojicoin-frontend --log-prefix none",
```
[the remark this answers is not in the document]
Comment (CRBl69) [reply src/typescript/package.json:28]: Not really.

I updated this in another PR, so I won't update it here again.

File: src/typescript/frontend/playwright.config.ts [hunk 3]
```diff
@@ -1,4 +1,9 @@
 import { defineConfig, devices } from "@playwright/test";
+import dotenv from "dotenv";
+import path from "path";
+
+// Read from ".env" file.
+dotenv.config({ path: path.resolve(__dirname, "../ci.env") });
```
Comment (xbtmatt) [inline src/typescript/frontend/playwright.config.ts:6]:
```suggestion
```

File: src/typescript/package.json (hunk 2 above)
Comment (xbtmatt) [inline src/typescript/package.json:28]:
```suggestion
    "e2e:frontend": "pnpm run load-env:e2e-frontend -- turbo run e2e:frontend --filter @econia-labs/emojicoin-frontend --log-prefix none",
```

File: src/typescript/package.json [hunk 4]
```diff
@@ -25,14 +25,13 @@
...
     "format": "turbo run format -- --write",
     "format:check": "turbo run format -- --check",
     "full-clean": "pnpm run clean && rm -rf node_modules && rm -rf sdk/node_modules && rm -rf frontend/node_modules",
     "lint": "turbo run lint",
     "lint:fix": "turbo run lint -- --fix",
     "load-env": "dotenv -e .env.local -e .env -e .env.example -e ../docker/example.local.env -e ../docker/.env",
-    "load-env:e2e-frontend": "dotenv -e .env.local -e .env -e .env.example -e ../docker/example.local.env -e ../docker/.env -e ci.env -v NODE_ENV=test",
```
Comment (xbtmatt) [inline src/typescript/package.json:35]: Add this back in and update it in the other PR when rebasing

**Review by xbtmatt — CHANGES_REQUESTED** [review]
This looks good overall- but we should really just remove the dotenv runtime stuff here and have your other PR rebase the changes that way we know it's fixed and we don't accidentally keep it in.

In general we should avoid adding anything into the repo that we don't plan on keeping in, it's just ripe for stale/dead code (I know I've done this a few times myself)

**Review by xbtmatt — APPROVED** [review]
Thanks for all the changes on this :) Looks great

**Pull request merged**